scripts/globusmonitor/globus_uploader_service.py

```diff
@@ -376,58 +386,69 @@ def notifyClowderOfCompletedTask(task):
-                                    "dataset_name": ds,
-                                    "dataset_id": dsid,
-                                    "action": "METADATA ADDED",
-                                    "metadata": datasetMD
-                                })
-                                updatedTask['contents'][ds]['files'][datasetMDFile]['metadata_loaded'] = True
-                                updatedTask['contents'][ds]['files'][datasetMDFile]['clowder_id'] = "attached to dataset"
-                                writeTaskToDatabase(updatedTask)
+                            dsmd = sess.post(clowder_host+"/api/datasets/"+dsid+"/metadata.jsonld",
+                                             headers={'Content-Type':'application/json'},
+                                             data=json.dumps(md))
+
+                            if dsmd.status_code in [104, 500, 502, 504]:
+                                logger.error("[%s] failed to attach metadata (%s: %s)" % (ds, dsmd.status_code, dsmd.text))
+                                updatedTask['contents'][ds]['files'][datasetMDFile]['retry'] = "%s: %s" % (dsmd.status_code, dsmd.text)
+                                response = "RETRY"
+                            elif dsmd.status_code != 200:
+                                logger.error("[%s] failed to attach metadata (%s: %s)" % (ds, dsmd.status_code, dsmd.text))
+                                updatedTask['contents'][ds]['files'][datasetMDFile]['error'] = "%s: %s" % (dsmd.status_code, dsmd.text)
+                                response = "ERROR"
                             else:
-                                # Remove metadata from activeTasks on success even if file upload fails in next step, so we don't repeat md
-                                logger.info("++ [%s] added metadata" % ds, extra={
-                                    "dataset_name": ds,
-                                    "dataset_id": dsid,
-                                    "action": "METADATA ADDED",
-                                    "metadata": datasetMD
-                                })
-                                del updatedTask['contents'][ds]['md']
-                                writeTaskToDatabase(updatedTask)
+                                if datasetMDFile:
+                                    logger.info("   [%s] added metadata from .json file" % ds, extra={
+                                        "dataset_name": ds,
+                                        "dataset_id": dsid,
+                                        "action": "METADATA ADDED",
+                                        "metadata": datasetMD
+                                    })
+                                    updatedTask['contents'][ds]['files'][datasetMDFile]['metadata_loaded'] = True
+                                    updatedTask['contents'][ds]['files'][datasetMDFile]['clowder_id'] = "attached to dataset"
```

Comment:
Not clear why this is an appropriate clowder_id value -- is this just convention?

Reply:
it's because we don't upload metadata.json files to clowder, we read them and attach them as metadata. this is just my way of saying "it got handled" when we don't have a clowder_id as I use absence of clowder_id elsewhere to check processing status. could replace it with dataset id it was attached to i guess.